File: 18/80-check-if-word-equals-summation-of-two-words/1880-check-if-word-equals-summation-of-two-words.py
class Solution:
    def isSumEqual(self, firstWord: str, secondWord: str, targetWord: str) -> bool:
        letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', ]
        firstValue, secondValue, targetValue = '', '', ''

        for char in firstWord:
            firstValue += str(letters.index(char))
        for char in secondWord:
            secondValue += str(letters.index(char))
        for char in targetWord:
            targetValue += str(letters.index(char))

        sum = int(firstValue) + int(secondValue)
        if sum == int(targetValue):
            return True
        else:
            return False


def main():

    work = Solution()
    work.isSumEqual(firstWord="aaa", secondWord="a", targetWord="aaaa")


if __name__ == '__main__':
    main()

File: 18/80-check-if-word-equals-summation-of-two-words/1880-check-if-word-equals-summation-of-two-words.js
/**
 * @param {string} firstWord
 * @param {string} secondWord
 * @param {string} targetWord
 * @return {boolean}
 */
var isSumEqual = function (firstWord, secondWord, targetWord) {
  let letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j'];
  let firstValue = '';
  let secondValue = '';
  let targetValue = '';

  for (let char of firstWord) {
    firstValue += letters.indexOf(char);
  }
  for (let char of secondWord) {
    secondValue += letters.indexOf(char);
  }
  for (let char of targetWord) {
    targetValue += letters.indexOf(char);
  }

  let sum = parseInt(firstValue) + parseInt(secondValue);
  if (sum === parseInt(targetValue)) {
    return true;
  } else {
    return false;
  }
};

console.log(
  isSumEqual((firstWord = 'acb'), (secondWord = 'cba'), (targetWord = 'cdb'))
);
